test/NuGet.Core.Tests/NuGet.Commands.Test/RestoreCommandTests.cs

```diff
@@ -1416,6 +1417,216 @@ public async Task RestoreCommand_CentralVersion_ErrorWhenCentralPackageVersionFi
+
+                var request = new TestRestoreRequest(spec, sources, pathContext.UserPackagesFolder, logger)
+                {
+                    LockFilePath = Path.Combine(projectPath, "project.assets.json"),
+                    ProjectStyle = ProjectStyle.PackageReference
+                };
+
+                var command = new RestoreCommand(request);
+
+                // Act
+                var result = await command.ExecuteAsync();
+
+                // Assert
+                Assert.False(result.Success);
+                var downgradeErrorMessages = logger.Messages.Where(s => s.Contains("Detected package downgrade: packageB from 2.0.0 to centrally defined 1.0.0.")).ToList();
+                Assert.Equal(1, downgradeErrorMessages.Count);
+            }
+        }
+
+        [Fact]
+        public async Task RestoreCommand_DowngradeIsNotErrorWhen_DowngradedByCentralDirectDependency()
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var logger = new TestLogger();
+                var projectName = "TestProject";
+                var projectPath = Path.Combine(pathContext.SolutionRoot, projectName);
+                var sources = new List<PackageSource> { new PackageSource(pathContext.PackageSource) };
+
+                var project1Json = @"
+                {
+                  ""version"": ""1.0.0"",
+                    ""restore"": {
+                                    ""projectUniqueName"": ""TestProject"",
+                                    ""centralPackageVersionsManagementEnabled"": true
+                    },
+                  ""frameworks"": {
+                    ""net472"": {
+                        ""dependencies"": {
+                                ""packageA"": {
+                                    ""version"": ""[2.0.0)"",
+                                    ""target"": ""Package"",
+                                    ""versionCentrallyManaged"": true
+                                },
+                                ""packageB"": {
+                                    ""version"": ""[1.0.0)"",
+                                    ""target"": ""Package"",
+                                    ""versionCentrallyManaged"": true
+                                }
+                        },
+                        ""centralPackageVersions"": {
+                            ""packageA"": ""[2.0.0)"",
+                            ""packageB"": ""[1.0.0)""
+                        }
+                    }
+                  }
+                }";
+
+                var packageA_Version200 = new SimpleTestPackageContext("packageA", "2.0.0");
+                var packageB_Version100 = new SimpleTestPackageContext("packageB", "1.0.0");
+                var packageB_Version200 = new SimpleTestPackageContext("packageB", "2.0.0");
+
+                packageA_Version200.Dependencies.Add(packageB_Version200);
+
+
```

Comment:
https://github.com/NuGet/NuGet.Client/blob/dev/docs/coding-guidelines.md Rule 7:

> Avoid more than one empty line at any time. For example, do not have two blank lines between members of a type.

Reply:
Removed the extra line.